File: v5configure_server/src/main/java/com/yunwen/android/v5configure_s/controller/LoginController.java
package com.yunwen.android.v5configure_s.controller;

import com.yunwen.android.v5configure_s.pojo.RequestResult;
import com.yunwen.android.v5configure_s.pojo.User;
import com.yunwen.android.v5configure_s.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.util.HtmlUtils;

import javax.servlet.http.HttpSession;

@Controller
public class LoginController {
    @Autowired
    UserService userService;

    @CrossOrigin
    @PostMapping(value = "/api/login")
    @ResponseBody
    public RequestResult login(@RequestBody User requestUser, HttpSession session) {
        String username = requestUser.getUsername();
        username = HtmlUtils.htmlEscape(username);

        User user = userService.get(username, requestUser.getPassword());
        if (null == user) {

            return new RequestResult(400);
        } else {
            session.setAttribute("user", user);
            return new RequestResult(200);
        }
    }

}

File: v5configure_server/src/main/java/com/yunwen/android/v5configure_s/controller/LoginController.kt
//package com.yunwen.android.v5configure_s.controller
//
//import com.yunwen.android.v5configure_s.pojo.LoginResult
//import com.yunwen.android.v5configure_s.pojo.User
//import com.yunwen.android.v5configure_s.service.UserService
//import org.springframework.beans.factory.annotation.Autowired
//import org.springframework.stereotype.Controller
//import org.springframework.web.bind.annotation.CrossOrigin
//import org.springframework.web.bind.annotation.PostMapping
//import org.springframework.web.bind.annotation.RequestBody
//import org.springframework.web.bind.annotation.ResponseBody
//import org.springframework.web.util.HtmlUtils
//
//
//@Controller
//class LoginController {
//    @Autowired
//    lateinit var userService: UserService
//    @CrossOrigin
//    @PostMapping(value = ["api/login"])
//    @ResponseBody
//     fun login(@RequestBody requestUser: User): LoginResult {
//
//
//         var username = requestUser.username
//         username = HtmlUtils.htmlEscape(username)
//      var  user = this.userService!![username, requestUser.password];
////         if (!Objects.equals("admin", username) || !Objects.equals("123456", requestUser.password)) {
////             var message = "账号密码错误";
////             System.out.println("test");
////             return LoginResult(400);
////         } else {
////             return LoginResult(200);
////         }
//        return if (null == user) {
//            LoginResult(400);
//        } else {
//            LoginResult(200);
//        }
//     }
//}